### PR TITLE
Catkin build test for commits

### LINKED ISSUE #107: Setup sanity checks for commits

### PR DESCRIPTION
**Pull request by aerin-brown-codes**
Resolved #107 

Creates a full ubuntu environment and runs catkin build on every commit or request to merge to the main branch. At the bottom of every such PR, the results of the workflow will be displayed.

### CODE REVIEW

File: .github/workflows/sanity-check.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: Integration Test
+
+on:
+  push:
+    branches:
+      - main
+      - CI
```
Comment (amotaouakkil123) [inline .github/workflows/sanity-check.yml:7]: What does CI stand for?

File: .github/workflows/sanity-check.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+name: Integration Test
+
+on:
```
Comment (amotaouakkil123) [inline .github/workflows/sanity-check.yml:3]: If I translate this correctly, this means that on a push to main or a pull request to main, this entire thing will run?

**Review by amotaouakkil123 — APPROVED** [review]

File: .github/workflows/sanity-check.yml (hunk 1 above)
Comment (aerin-brown-codes) [reply .github/workflows/sanity-check.yml:7]: Continuous Integration. I needed to make sure the workflow would run on the branch I was working on so I could confirm it worked lol

File: .github/workflows/sanity-check.yml (hunk 2 above)
Comment (aerin-brown-codes) [reply .github/workflows/sanity-check.yml:3]: Exactly! Pull requests will have the result of the workflow run at the bottom down near the merge button, and the push requests will have a little checkmark or X next to them, if I'm remembering the placement right. Either way the results of every workflow run can be found in the Actions tab.

**Review by JosephSaliba01 — APPROVED** [review]

File: .github/workflows/sanity-check.yml (hunk 1 above)
Comment (JosephSaliba01) [reply .github/workflows/sanity-check.yml:7]: @amotaouakkil123 We need to resolve this conversation and merge if you don't have anything else to ask

File: .github/workflows/sanity-check.yml (hunk 2 above)
Comment (JosephSaliba01) [reply .github/workflows/sanity-check.yml:3]: @amotaouakkil123 need to resolve for merge

**Pull request merged**